src/vm.rb

```diff
@@ -61,6 +61,61 @@ def destroy
 
     private
 
+    CERTS_DIR = '/var/lib/metalware/certs'
+
+    def libvirt_host
+      node.config.libvirt_host
+    end
+
+    def libvirt
+      Libvirt.open("qemu://#{libvirt_host}/system")
+    end
+
+    def certificate?
+      File.exist?("#{CERTS_DIR}/#{libvirt_host}-key.pem")
+    end
+
+    def generate_certificate
+      generate_certificate_key
+      generate_certificate_info
+      generate_server_certificates
+      puts certificate_generation_message
+      exit
+    end
+
+    def generate_certificate_key
+      SystemCommand.run("certtool --generate-privkey > #{CERTS_DIR}/#{libvirt_host}-key.pem")
+    end
+
+    def generate_certificate_info
+      File.open("#{CERTS_DIR}/#{libvirt_host}.info", 'w') do |f|
+        f.puts 'organization = Alces Software'
+        f.puts "cn = #{libvirt_host}"
+        f.puts 'tls_www_server'
+        f.puts 'encryption_key'
+        f.puts 'signing_key'
+      end
+    end
+
+    def generate_server_certificates
+      SystemCommand.run("certtool --generate-certificate \
+                        --load-privkey #{CERTS_DIR}/#{libvirt_host}-key.pem \
+                        --load-ca-certificate #{CERTS_DIR}/cacert.pem \
+                        --load-ca-privkey #{CERTS_DIR}/cakey.pem \
+                        --template #{CERTS_DIR}/#{libvirt_host}.info \
+                        --outfile #{CERTS_DIR}/#{libvirt_host}-cert.pem")
+    end
+
+    def certificate_generation_message
+      <<-EOF
+  The following certificates have been generated, copy them
+  to the specified remote directory on #{libvirt_host}:
+
+    from (local): #{CERTS_DIR}/#{libvirt_host}-{key,cert}.pem
+     to (remote): /etc/pki/libvirt/{servercert.pem,/private/serverkey.pem}
+      EOF
```

Comment:
The formatting on this heredoc looks slightly off - typically you would indent the text 1 level (i.e. 2 spaces for us) from the `<<-EOF`, and in this case you might want the `from` and `to` lines to be indented 1 level further, or at least they should line up with each other.

Also, because of the preceding whitespace on each line the output text will appear unnecessarily indented. There's a method in `active_support`, which we already have as a dependency, which will strip off the shortest level of whitespace of all lines. Just replace the first line with `<<-EOF.strip_heredoc` (you will probably also need to add a `require 'active_support/core_ext/string/strip'` up the top so this method is available).